### DISCUSSION TITLE
DID Auth

### DISCUSSION THREAD

**Discussion by atz3n** [decentralized-identity/veramo#711] [General]
Hello,

I just wanted to ask if there is a way to verify a DID with Veramo like the [DID Auth](https://github.com/WebOfTrustInfo/rwot6-santabarbara/blob/master/final-documents/did-auth.md) Protocol. Some challenge - response mechanism.

**Comment by awoie** [comment]
> Hello,
> 
> I just wanted to ask if there is a way to verify a DID with Veramo like the [DID Auth](https://github.com/WebOfTrustInfo/rwot6-santabarbara/blob/master/final-documents/did-auth.md) Protocol. Some challenge - response mechanism.

You could use VP-Auth for this. Essentially, creating a VP without a VC that just contains a signature of the DID of the holder + some nonce + some domain to prevent replay attacks.

Other protocols exist, such as SIOPv2 but we haven't implemented support yet.

**Reply by atz3n** [reply]
👍  thx
